R/fmelt.R

```diff
@@ -200,6 +200,9 @@ melt.data.table = function(data, id.vars, measure.vars, variable.name = "variabl
       measure.vars = eval.result
     }
   }
+  if (length(measure.vars) == length(unlist(measure.vars))) {
```

Comment:
`unlist()` is relatively expensive. We should only do it within the branch if possible (and definitely not twice).

Is this the same as `all(vapply_1i(measure.vars, length) == 1L)`? Or is it really `sum(vapply_1i(measure.vars, length)) == length(measure.vars)`? (the latter would apply if some elements could be empty, others length-2, etc)